### PR TITLE
add multi-argument gcd and lcm

### LINKED ISSUE #89: Add multiple-argument gcd, lcm, patterned on `g/*` and friends
This is how the operation is defined in Scheme:

```scheme
(define (g:gcd:n args)
  (cond ((null? args) :zero)
        ((null? (cdr args)) (car args))
        (else
         (let lp
             ((as (cddr args))
              (ans (g:gcd:bin (car args) (cadr args))))
           (cond ((null? as) ans)
                 ((g:one? ans) ans)
                 (else
                  (lp (cdr as) (g:gcd:bin ans (car as)))))))))
```

The task here would be to

- rename the generics now to something like `g/-gcd` and `g/-lcm`
- expose a multi-arg `g/gcd`, `g/lcm` that wraps them and does this trick with both functions.

### LINKED ISSUE #177: Add multiple-argument gcd, lcm, patterned on `g/*` and friends
This is how the operation is defined in Scheme:

```scheme
(define (g:gcd:n args)
  (cond ((null? args) :zero)
        ((null? (cdr args)) (car args))
        (else
         (let lp
             ((as (cddr args))
              (ans (g:gcd:bin (car args) (cadr args))))
           (cond ((null? as) ans)
                 ((g:one? ans) ans)
                 (else
                  (lp (cdr as) (g:gcd:bin ans (car as)))))))))
```

The task here would be to

- `rename the generics now to something like `g/-gcd`, `g/-lcm`
- expose a multi-arg `g/gcd`, `g/lcm` that wraps them and does this trick with both functions.

### PR DESCRIPTION
**Pull request by kolharsam**
this PR should resolve sicmutils/placeholder#88

### CODE REVIEW

**Comment by kolharsam** [comment]
@sritchie I have made the changes for `gcd` alone now. I'd like to clean up the mistakes that I've made here and then based on that would like to do further work on `lcm`. I hope that is okay?

**Comment by sritchie** [comment]
Hey @kolharsam ! It's probably worth doing a quick chat on Clojurians Slack if you're on that forum so we can go over what this PR is supposed to achieve! Send me a note in https://clojurians.slack.com/archives/C01ECA9AA74 via the username @sritchie and we'll discuss. Sound good?

File: src/sicmutils/generic.cljc [hunk 1]
```diff
@@ -447,10 +447,27 @@
 (defmethod remainder :default [n d]
   (remainder-default n d))
 
+;;; FIXME?: is the multi-argument definition required? [[gcd]], I don't fully
+;;; understand what [[add]] is doing. Where do I look?
 (defgeneric gcd 2
```
Comment (sritchie) [inline src/sicmutils/generic.cljc:452]: Ah! So the idea is that this is a generic function already, and many different types extend it. But it only works for two arguments. So, the task is:

- change this to `(defgeneric -gcd 2 ...`)
- delete the implementation you have on line `458`
- change `465` to call `-gcd` internally

Then the exposed `gcd` will be multiple arity. We'll need tests too. Let me know if that makes sense!

**Comment by sritchie** [comment]
@kolharsam , closing this since it looks like it is abandoned for now. Thank you!

**Pull request closed without merging**